identity/src/main/java/com/stripe/android/identity/navigation/SelfieFragment.kt

```diff
@@ -44,62 +39,52 @@ internal class SelfieFragment(
     identityViewModelFactory: ViewModelProvider.Factory
 ) : IdentityCameraScanFragment(identityCameraScanViewModelFactory, identityViewModelFactory) {
     override val fragmentId = R.id.selfieFragment
-    override val shouldObserveDisplayState = true
-
-    private lateinit var flashAnimatorSet: AnimatorSet
-    private lateinit var binding: SelfieScanFragmentBinding
-    private lateinit var continueButton: LoadingButton
-    private lateinit var messageView: TextView
-    private lateinit var flashMask: View
-    private lateinit var scanningView: CardView
-    private lateinit var resultView: ScrollView
-    private lateinit var padding: View
-    private lateinit var capturedImages: RecyclerView
-    private lateinit var allowImageCollection: CheckBox
-
-    internal var flashed = false
-
-    @VisibleForTesting
-    internal val selfieResultAdapter: SelfieResultAdapter = SelfieResultAdapter()
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View {
-        binding = SelfieScanFragmentBinding.inflate(inflater, container, false)
-        cameraView = binding.cameraView
-        continueButton = binding.kontinue
-        messageView = binding.message
-        flashMask = binding.flashMask
-        scanningView = binding.scanningView
-        resultView = binding.resultView
-        padding = binding.padding
-        capturedImages = binding.capturedImages
-        allowImageCollection = binding.allowImageCollection
-        capturedImages.adapter = selfieResultAdapter
-
-        flashAnimatorSet = AnimatorSet().apply {
-            playSequentially(
-                ObjectAnimator.ofFloat(flashMask, "alpha", FLASH_MAX_ALPHA).setDuration(
-                    FLASH_ANIMATION_TIME
-                ),
-                ObjectAnimator.ofFloat(flashMask, "alpha", 0f).setDuration(
-                    FLASH_ANIMATION_TIME
-                )
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            val changedDisplayState by identityScanViewModel.displayStateChangedFlow.collectAsState()
+            val newDisplayState by remember {
+                derivedStateOf {
+                    changedDisplayState?.first
+                }
+            }
```

Comment:
maybe we can make this more readable:
```suggestion
            val (newState, prevState) by identityScanViewModel.displayStateChangedFlow.collectAsState()
```

Reply:
hmm seems not doable on a mutable flow, got this error: `Initializer required for destructuring declaration`, tried to make `dislplayStateChangedFlow` to type `MutableStateFlow<Pair<IdentityScanState, IdentityScanState>>` but still got similar error